manifests/install/rabbitmqadmin.pp

```diff
@@ -40,9 +40,24 @@
     }
 
     if !($rabbitmq::use_config_file_for_plugins) {
-      $rabbitmqadmin_archive_require = [Class['rabbitmq::service'], Rabbitmq_plugin['rabbitmq_management']]
+      $rabbitmqadmin_archive_require = [
+        Class['rabbitmq::service'],
+        Rabbitmq_plugin['rabbitmq_management'],
+        Exec['remove_old_rabbitmqadmin_on_upgrade']
+      ]
     } else {
-      $rabbitmqadmin_archive_require = [Class['rabbitmq::service'], File['enabled_plugins']]
+      $rabbitmqadmin_archive_require = [
+        Class['rabbitmq::service'],
+        File['enabled_plugins'],
+        Exec['remove_old_rabbitmqadmin_on_upgrade']
+      ]
+    }
+
+    Exec { 'remove_old_rabbitmqadmin_on_upgrade':
+      path        => ['/bin','/usr/bin','/sbin','/usr/sbin'],
+      command     => "rm ${rabbitmq::rabbitmq_home}/rabbitmqadmin",
```

Comment:
Could we do this with a file resource with `ensure => absent` vs. using a shell exec (but still using the same `onlyif`)?

Reply:
I am aware, that exec resources should be avoided, if there is any possibilty to fullfil the task as well with other basically idempotent resources. In order to provide idempotence for this exec resource, `onlyif` and `refreshonly` parameter are set. 

Managing a file resource presence dependent on the refresh signal seems not to be possible imho.